### ISSUE TITLE
Polynomial learning rate callback fails on setup

### ISSUE THREAD

**Issue by oyamay** [LLNL/lbann#1581]
When a polynomial decay callback is used, it fails on [`poly_learning_rate::setup`](https://github.com/LLNL/lbann/blob/21d63e356dc51ecb2a600042a4b732dbef783bde/src/callbacks/learning_rate.cpp#L269-L274) to try to get the number of iterations per epoch via the training execution context, which is not set on startup. 
There needs to be another way to get information about the training dataset without execution contexts.
